### PR TITLE
Allow objects with non-existing keys in PartialDeep

### PR DESCRIPTION
**Pull request by merlinnot**
This fixes the following issue (does not compile with latest TypeScript):

```ts
interface Interface {
  a: 'a',
}

type PartialInterface = PartialDeep<Interface>;

const value: PartialInterface = {};
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
@kainiedziela Can you review?

**Comment by kainiedziela** [comment]
I'd say this warrants an extra test case to cover empty objects then.

File: test-d/partial-deep.ts [hunk 1]
```diff
@@ -39,8 +39,8 @@ expectType<undefined>(partialDeepFoo.bar!.undefined);
 expectType<Map<string | undefined, string | undefined> | undefined>(partialDeepFoo.bar!.map);
 expectType<Set<string | undefined> | undefined>(partialDeepFoo.bar!.set);
 expectType<Array<string | undefined> | undefined>(partialDeepFoo.bar!.array);
-expectType<['foo' | undefined] | undefined>(partialDeepFoo.bar!.tuple);
+expectType<[('foo')?] | undefined>(partialDeepFoo.bar!.tuple);
```
Comment (kainiedziela) [inline test-d/partial-deep.ts]: Any particular reason for parentheses here? You don't add them in the `readonly tuple` test (line 46).

**Comment by merlinnot** [comment]
I removed extra parentheses, I'm used to Prettier doing it for me, had to disable it for this repo and didn't notice it.

As for the test case, maybe my example was a little too minimal. Here's another one:

```ts
interface Interface {
  a: 'a',
  b: 'b',
}

type PartialInterface = PartialDeep<Interface>;

// Does not compile.
const value: PartialInterface = { a: 'a' };
```

**Comment by merlinnot** [comment]
I'm not really used to test tools that you guys use, I checked the repositories and it seems it's deprecated, so I wasn't exactly sure what's your approach to this.

I'd appreciate any guidance on what tests would you expect, I'll try to add these.

**Comment by sindresorhus** [comment]
We use https://github.com/SamVerschueren/tsd. It's not deprecated.

**Comment by merlinnot** [comment]
Oh, I thought that's the one: https://github.com/DefinitelyTyped/tsd. First result in Google :)

Thanks for the link, I'll try to add some tests tomorrow.

**Comment by kainiedziela** [comment]
I'd say you can just follow the pattern already established in the present type-fest test cases. The library used is pretty intuitive.

**Comment by merlinnot** [comment]
@kainiedziela I sent you an invite. I was playing with it a little but the only test case I could come up with is to basically paste the code above to the test file, which I guess is not what's expected.

I also changed the existing tests to make them pass, so it's already a regression test of sorts.

**Comment by kainiedziela** [comment]
No, I think that would be quite intuitive. I pushed the test cases that should cover our needs. I also removed the version change as I believe that should be handled by package maintainers.

@sindresorhus this looks good to go.

**Comment by sindresorhus** [comment]
Thanks for fixing :)
